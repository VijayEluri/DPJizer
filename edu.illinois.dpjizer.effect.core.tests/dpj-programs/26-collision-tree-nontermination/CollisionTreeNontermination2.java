/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
class CT<region P> {
	region L, R;

	protected CT<P:L> l in P:L;
	protected CT<P:R> r in P:R;

	public CT() {
	}

	public <region Q>void intersect(CT<Q> collisionTree) 
	{
		collisionTree.intersect(l);
		collisionTree.<region P:R>intersect(r);
	}
}
