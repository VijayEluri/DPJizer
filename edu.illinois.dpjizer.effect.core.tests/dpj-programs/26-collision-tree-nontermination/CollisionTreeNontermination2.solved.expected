CT.<>intersect(CT<Q>) reads P : CT.L : *, P : CT.R : *, Q : CT.L : *, Q : CT.R : *
CT.CT() pure
